Sources/StreamChatUI/Composer/ComposerVC.swift

```diff
@@ -796,10 +872,36 @@ open class ComposerVC: _ViewController,
             log.assertionFailure(error.localizedDescription)
         }
     }
+    
+    open func checkChannelCooldown() {
+        if currentCooldownTime() > 0 {
+            cooldownTracker.start(with: currentCooldownTime())
+        }
+    }
+    
+    func currentCooldownTime() -> Int {
+        guard let cooldownDuration = channelController?.channel?.cooldownDuration,
+              let currentUserLastMessage = channelController?.channel?.lastMessageFromCurrentUser else {
+            return 0
+        }
+        
+        let currentTime = Date().timeIntervalSince(currentUserLastMessage.createdAt)
+        return cooldownDuration - Int(currentTime)
+    }
```

Comment:
This function should actually be moved to `ChatChannelController` so that we can use this in the SwiftUI SDK as well 🙂 CC @martinmitrevski

Reply:
Moved to ChatChannelController